### ISSUE TITLE
support screen reader reading the line and audio cues when go to next/previous diff are used in diff view

### ISSUE THREAD

**Issue by meganrogge** [microsoft/vscode#171200]
Fulfilled for normal editors in these:
https://github.com/microsoft/vscode/issues/166518
https://github.com/microsoft/vscode/issues/164988

**Comment by meganrogge** [comment]
For deleted lines, we'll not select anything so nothing will get read

**Comment by isidorn** [comment]
Take into account that we support F7 and Shift+F7 as an accessible way to navigate across diffs.
Maybe this is not the best us of shortcuts, but I am not sure if we need to add additional functionality here.

**Comment by jooyoungseo** [comment]
@isidorn I have some second thoughts. I don't think it is a completely new functionality because Alt+F5 and Alt+Shift+F5 shortcut keys have already been working even for screen users as a general feature regardless of the F7/Shift+F7 alternatives. However, the issue was that the existing feature (Alt+F5) was not reading the currently focused line and there was no contextual info on whether it is added or deleted line for screen readers. Now that there is a way to improve this feature, I think there is no reason to prevent screen reader users from taking the advantage of this general feature in a more accessible way. FYI, people can keep using F7 navigation if they would like, but F7 view only gives few lines of code before and after any changes so screen reader users will have to switch back and forth between the alternative F7 diff view and diff view editor in order to understand the code context. Opening the dorr for F5 can alleviate this issue. Last but not least, it seems like this new addition does not necessarily introduce any conflicts against the existing F7 methods. Diffing has been one of the accessibility challenges among blind programmers. I believe this improvement can open another door for them (and us) in conjunction with F7.

**Comment by meganrogge** [comment]
@jooyoungseo I see your point - we are merging the change in, so you should be able to try it out on Monday.

**Comment by jooyoungseo** [comment]
@meganrogge  -- I appreciate it! I will test this once it's available in insiders.

**Comment by jooyoungseo** [comment]
@meganrogge -- This works so nicely! Thank you so much for this meaningful improvement. BTW, I have found that audio cues does not work from time to time in the following circumstances. Let's say you have multiple lines of changes (line A; line B; line C, for example) and you can move between changes by pressing Alt+F5/Alt+Shift+F5. Sometimes, audio cues does not work on the line B. I experienced this issue when I press Alt+F5 multiple times in one sec interval. However, I cannot exactly narrow down the issue and it is hard for me to provide any reproducible steps. Just noticed that the audio cues are not working occasionally.

**Comment by meganrogge** [comment]
@jooyoungseo thrilled to hear that it's working nicely for you. That is happening as designed because we don't allow audio cues to play if they're already playing - here's the issue describing that. #164921 Something we could consider supporting is a queue that takes in any requests to play a sound and plays it when the prior one finishes. What do you think of that? Also @hediet is the owner of audio cues, so perhaps he has some thoughts on this.

**Comment by jooyoungseo** [comment]
@meganrogge 

> Something we could consider supporting is a queue that takes in any requests to play a sound and plays it when the prior one finishes. What do you think of that?

Wouldn't it introduce any audio lags? Is this something like `aria-live = "polite"`? If I am quickly swiping through in the diff editor via arrow keys, I would expect the audio cue events of currently focused line override any pending or existing audio cues just like `aria-live = "assertive"`. If we cannot make one-size-fits-all solution, making this as a custom user setting would be another fair compromise.

**Comment by meganrogge** [comment]
I think for go to next/ previous difference, it makes sense to let the cues play because they won't overlap badly like the task ones do. I'll make an issue and that fix
